### PR TITLE
Add macos and window platform in CI workflow

### PR DESCRIPTION
**Pull request by heemin32**
Signed-off-by: Heemin Kim <heemin@amazon.com>

### Description
Add macos and window platform in CI workflow

### Issues Resolved
https://github.com/opensearch-project/dashboards-maps/issues/55

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/unit-tests-workflow-for-customImportMap.yml [hunk 1]
```diff
@@ -2,21 +2,32 @@ name: Unit tests workflow
 on:
```
Comment (junqiu-lei) [inline .github/workflows/unit-tests-workflow-for-customImportMap.yml:2]: In my understand, this workflow only related to customImportMap plugin, which plugin is under `src/plugins/custom_import_map` path, can we also specify a path condition check? In that case, other maps plugins change under `src/plugins/` won't trigger this workflow, the code could be:
```
on:
  push:
    branches:
      - "*"
      - "feature/**"
    paths:
      - src/plugins/custom_import_map/**
  pull_request:
    branches:
      - "*"
      - "feature/**"
    paths:
      - src/plugins/custom_import_map/**
```

Comment (heemin32) [reply .github/workflows/unit-tests-workflow-for-customImportMap.yml:2]: Other plugins won't trigger this workflow because they are in their own repo.

Comment (heemin32) [reply .github/workflows/unit-tests-workflow-for-customImportMap.yml:2]: Talked with @junqiu-lei offline. We are going to add another plugin in this repo. https://github.com/opensearch-project/dashboards-maps/tree/feature/new-maps/src/plugins/maps_dashboards

So, restricting the github action to the specific folder.

Comment (junqiu-lei) [reply .github/workflows/unit-tests-workflow-for-customImportMap.yml:2]: There have another developing maps visualization plugin in this repo https://github.com/opensearch-project/dashboards-maps/tree/feature/new-maps/src/plugins/maps_dashboards. Just also offline synced with Heemin.

**Review by junqiu-lei — APPROVED** [review]
LGTM, thank you!

**Review by naveentatikonda — APPROVED** [review]

**Review by VijayanB — APPROVED** [review]

**Pull request merged**